tests/test_srcsetparse.py

```diff
@@ -0,0 +1,335 @@
+# Copyright (C) 2022 Stefan fisk
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+Test srcset attribute parsing.
+"""
+
+from dataclasses import dataclass
+from html.entities import html5
+import pytest
+import re
+from linkcheck.htmlutil.srcsetparse import parse_srcset
+
+@dataclass
+class Spec:
+    srcset: str
+    expect: str
+    description: str = None
+    resolve: bool = False
+
+# Adapted from https://github.com/web-platform-tests/wpt/blob/master/html/semantics/embedded-content/the-img-element/srcset/parse-a-srcset-attribute.html
```

Comment:
I'm not sure how licensing works for this. WebKit for example just keep a full copy of the test kit in their repo https://github.com/WebKit/WebKit/tree/8afe31a018b11741abdf9b4d5bb973d7c1d9ff05/LayoutTests/imported/w3c/web-platform-tests.

Reply:
Here's another commit where the original WPT file is retained but with the copyright and license added to the top. The file is then parsed when running the tests.

https://github.com/stefanfisk/linkchecker/commit/255d49f41b0891201074b38f665d3ed0f8bee233